Flaky integration test: `test_override_build_in_commands`

Nodeid: `tests_integration/plugin/test_override_by_external_plugins.py::test_override_build_in_commands`
# 🟢 Setup passed
# 🔴 Call failed
```python
runner = <tests_integration.conftest.SnowCLIRunner object at 0x7fc723332020>
test_root_path = PosixPath('/home/runner/work/snowflake-cli/snowflake-cli/tests_integration')
_install_plugin = None
caplog = <_pytest.logging.LogCaptureFixture object at 0x7fc723333c10>

    @pytest.mark.integration
    def test_override_build_in_commands(runner, test_root_path, _install_plugin, caplog):
        config_path = (
            test_root_path / "config" / "plugin_tests" / "override_plugin_config.toml"
        )
    
        result = runner.invoke(["--config-file", config_path, "connection", "list"])
    
>       assert (
            "Cannot register plugin [override]: Cannot add command [snow connection list] because it already exists."
            in caplog.messages
        )
E       AssertionError: assert 'Cannot register plugin [override]: Cannot add command [snow connection list] because it already exists.' in ['Creating directory /tmp/tmp_d8gx_4r/.snowflake/logs']
E        +  where ['Creating directory /tmp/tmp_d8gx_4r/.snowflake/logs'] = <_pytest.logging.LogCaptureFixture object at 0x7fc723333c10>.messages

tests_integration/plugin/test_override_by_external_plugins.py:28: AssertionError
```
https://github.com/snowflakedb/snowflake-cli/blob/07f630bacaeae547b7588a8ec0bcc2362fa8820d/tests_integration/plugin/test_override_by_external_plugins.py#L28
# 🟢 Teardown passed

+1
```python
AssertionError: assert 'Cannot register plugin [override]: Cannot add command [snow connection list] because it already exists.' in ['Creating directory /tmp/tmp_d8gx_4r/.snowflake/logs']
 +  where ['Creating directory /tmp/tmp_d8gx_4r/.snowflake/logs'] = <_pytest.logging.LogCaptureFixture object at 0x7fc723333c10>.messages
```

--- a/src/snowflake/cli/_app/commands_registration/command_plugins_loader.py
+++ b/src/snowflake/cli/_app/commands_registration/command_plugins_loader.py
@@ -69,7 +69,20 @@ class CommandPluginsLoader:
                 )
 
     def load_all_registered_plugins(self) -> List[LoadedCommandPlugin]:
-        for plugin_name, plugin in self._plugin_manager.list_name_plugin():
+        all_plugins = list(self._plugin_manager.list_name_plugin())
+        builtin_plugin_names = set(get_builtin_plugin_name_to_plugin_spec().keys())
+
+        def plugin_sort_key(name_plugin_tuple):
+            _plugin_name, _ = name_plugin_tuple
+            is_builtin = _plugin_name in builtin_plugin_names
+            return (
+                not is_builtin,
+                _plugin_name,
+            )
+
+        sorted_plugins = sorted(all_plugins, key=plugin_sort_key)
+
+        for plugin_name, plugin in sorted_plugins:
             self._load_plugin(plugin_name, plugin)
         return list(self._loaded_plugins.values())
 

